### ISSUE TITLE
KSampler Efficient - Shouldn't Randomize be applied automatically?

### ISSUE THREAD

**Issue by BlinkerHigh** [jags111/efficiency-nodes-comfyui#205]
Hello,
KSampler Efficient randomize function isnt applied automatically, is this the standard behavior?
If I don't press the randomize button myself, the seed will remain the same each generation.

If it is the desired behavior, is there a way to randomize it automatically?

**Comment by jags111** [comment]
there is a small triangle in between which is a widget which needs to be pressed.

**Comment by jags111** [comment]
when you want randomize activated click like shown
<img width="765" alt="2024-06-18_13-06-13" src="https://github.com/jags111/efficiency-nodes-comfyui/assets/5968619/7ae46a24-2146-408c-bcd3-32d16eb6a22d">

**Comment by BlinkerHigh** [comment]
Well, for some reason it doesn't work.
When pressing the green icon with arrows, I see a -1 appear but it disappears right away and is replaced with a seed number.

**Comment by BlinkerHigh** [comment]
![image](https://github.com/jags111/efficiency-nodes-comfyui/assets/121453198/85b93b9b-c361-47b5-9dbd-8fb282a9a0a3)

**Comment by BlinkerHigh** [comment]
I have the latest version by the way.

Even when I enter -1 manually, a seed is automatically applied.
That's why I was wondering if this is the intended behavior or not but it won't let me go with -1.

**Comment by BlinkerHigh** [comment]
Looking at your screenshot, my node is different from yours

![image](https://github.com/jags111/efficiency-nodes-comfyui/assets/121453198/0582e100-5cd6-4f52-aa47-8ef6ea34ed82)

**Comment by nadora35** [comment]
same issue

**Comment by openxbmc** [comment]
same issus. but maybe another node conflict. because I found this issue yesterday. when I install some nodes

**Comment by BlinkerHigh** [comment]
> same issus. but maybe another node conflict. because I found this issue yesterday. when I install some nodes

Which one did you install? How to find the conflicting nodes?

**Comment by jags111** [comment]
please note -1 is not the seed and when you press run a new seed will appear. and every time it will get random seed.. if the button is active. you can also press the left button to activate a new seed.

**Comment by nadora35** [comment]
@jags111  what is this ... plz fix this bus

https://github.com/jags111/efficiency-nodes-comfyui/assets/57017061/2c5c7cee-c775-4645-b75f-2050896b0c24

**Comment by openxbmc** [comment]
> > same issus. but maybe another node conflict. because I found this issue yesterday. when I install some nodes
> 
> Which one did you install? How to find the conflicting nodes?

sorry, I installed lots of nodes. I tried to unsitall some but nothing change.so maybe conflict with new ComfyUI update?

**Comment by openxbmc** [comment]
> please note -1 is not the seed and when you press run a new seed will appear. and every time it will get random seed.. if the button is active. you can also press the left button to activate a new seed.

I understand what you mean, normally, if we queue to execute the task, the condition of random seed is different for every picture, but the problem we are having is that the queue is executing the task even with a random seed, but the resulting pictures are all the same.

What I am curious about at the moment is why it worked well before and now it doesn't, I just installed some new nodes and then upgraded ComfyUI to the latest version.

**Comment by jags111** [comment]
let me again re run the workflow test and update with 2 random seeds.

**Comment by creepysnake** [comment]
Same issue, in my case, after removing the cg-use-everywhere (a tool for hidden lines between nodes)
the random function works properly.

**Comment by openxbmc** [comment]
> Same issue, in my case, after removing the cg-use-everywhere (a tool for hidden lines between nodes) the random function works properly.

 I checked nodes but not found this nodes, maybe something else.

**Comment by jags111** [comment]
most of the random seed issues stem not from code but from the Java script backend conflicting with the performance of the seed change behaviour. This bug originates from any java script oriented installs from any nodes installs or updates and will change from browser to browser and also due to system dependencies changing. While it works in most cases some times it will error due to java script backend not up to date. 
I will see how the change is protected in eff codes and will give a feedback. if it persists in run make an external seed change node and plug same in.

**Comment by BlinkerHigh** [comment]
Thanks for looking into it :)

**Comment by ricperry** [comment]
I wonder if this should be reclassified officially as a bug. This problem makes this node useless for my workflow, even though I really want to use it to simplify things.

**Comment by BlinkerHigh** [comment]
> I wonder if this should be reclassified officially as a bug. This problem makes this node useless for my workflow, even though I really want to use it to simplify things.

I tend to agree since other nodes with this randomize function work as expected.
For instance, the Ultimate SD upscaler node doesn't have this problem.

**Comment by BlinkerHigh** [comment]
Any update?

**Comment by BlinkerHigh** [comment]
Any update? Still doesnt work.
